eth-node/bridge/geth/envelope.go

```diff
@@ -2,29 +2,112 @@ package gethbridge
 
 import (
 	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/waku"
 	"github.com/status-im/status-go/whisper/v6"
 )
 
 type gethEnvelopeWrapper struct {
-	envelope *whisper.Envelope
+	shhEnvelope  *whisper.Envelope
+	wakuEnvelope *waku.Envelope
 }
```

Comment:
Wouldn't it make more sense to create an `type Envelope interface` defining what methods `waku.Envelop` and `whisper.Envelope` should implement to be of that type?

Reply:
There is such a thing: https://github.com/status-im/status-go/blob/develop/eth-node/types/envelopes.go#L5-L8

Reply:
I see.